plugin/fzf.vim

```diff
@@ -632,10 +632,13 @@ function! s:execute(dict, command, use_height, temps) abort
       return []
     endif
   elseif has('win32unix') && $TERM !=# 'cygwin'
-    let shellscript = s:fzf_tempname()
-    call writefile([command], shellscript)
-    let command = 'cmd.exe /C '.fzf#shellescape('set "TERM=" & start /WAIT sh -c '.shellscript)
-    let a:temps.shellscript = shellscript
+    let cygwin_version = substitute(system("uname -r |grep --only-matching '^[0-9.]\\+' |tr -d '.'"), '\_s*$', '', '')
```

Comment:
Is this safe to cache?

Reply:
Introduced local on-demand caching.
Good point, I assumed `win32unix` means it's Cygwin, need to look into this.

Reply:
Added an explicit check for Cygwin